### PR TITLE
Fixing UserPlugins.defineFlag

### PR DESCRIPTION
**Pull request by john-science**
## Description

This is in response to #1191.  @onufer I was able to reproduce your bug locally (though using a stand-alone App), and this fixed the bug for me.  Before we merge this PR I was hoping you would get to run this commit/change through your workflow to make sure this fixes the issue for you as well.  Thanks!

---

## Checklist

- [X] This PR has only one purpose or idea.
- [X] Tests have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [X] The documentation is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

**Review by bsculac — APPROVED** [review]

**Review by keckler — APPROVED** [review]
So this appears to have just been an oversight before, not really a "bug"? We just forgot to actually load the flags when we imported the user plugins?

Either way, looks fine now.

File: armi/apps.py [hunk 1]
```diff
@@ -332,6 +337,11 @@ def __registerUserPluginsInternalImport(self, pluginPath):
         assert issubclass(plugin, plugins.UserPlugin)
         self._pm.register(plugin)
 
+        # ensure UserPlugin flags are loaded
+        newFlags = plugin.defineFlags()
+        if newFlags:
+            Flags.extend(newFlags)
+
```
Comment (onufer) [inline armi/apps.py:344]: this works. there is a bit of duplicate code here, so maybe eventually we should make a `registerUserPlugin` method where everything from 
assert issubclass(plugin, plugins.UserPlugin)
down goes in

File: armi/apps.py [hunk 2]
```diff
@@ -35,8 +35,9 @@
 
 from armi import context, plugins, pluginManager, meta, settings
 from armi.reactor import parameters
-from armi.settings import Setting
+from armi.reactor.flags import Flags
 from armi.settings import fwSettings
+from armi.settings import Setting
```
Comment (onufer) [inline armi/apps.py:40]: appears unused?

**Review by onufer — APPROVED** [review]

File: armi/apps.py (hunk 2 above)
Comment (onufer) [reply armi/apps.py:40]:
```suggestion
```

Comment (john-science) [reply armi/apps.py:40]: No, this import is needed for line 121.

**Pull request merged**